src/main/java/org/jboss/tools/intellij/openshift/utils/odo/ComponentSourceType.java

```diff
@@ -29,11 +27,14 @@ public String toString() {
     }
 
     public static ComponentSourceType fromAnnotation(String annotation) {
-        for(ComponentSourceType sourceType : values()) {
-            if (sourceType.annotation.equals(annotation)) {
-                return sourceType;
-            }
+        switch (annotation) {
```

Comment:
I think the previous pattern was more extensible, what's the rationale for this change ?

Reply:
I did some extra refactoring while creating ComponentKind enum.. reverted.